Sources/Client/Client/Client.swift

```diff
@@ -191,3 +226,35 @@ extension Client {
         case delete = "DELETE"
     }
 }
+
+// MARK: - Retry Requester
+
+/// A retry requester is a helper protocol for implementing retry request logic.
+public protocol ClientRetryRequester {
```

Comment:
This whole protocol is confusing to me..
* In the example, why do we send `completion` into a call for `Client.shared.request`? Shouldn't `completion` be called as soon as connection is available? So it should be:
```
final class RetryRequester: ClientRetryRequester {
    func reconnectForExpiredToken<T: Decodable>(endpoint: Endpoint, _ completion: @escaping Client.Completion<T>) {
        myReconnect {
          self.connectedWithNewToken()
          completion()
        }
    }
}
```
* It's confusing why the user has to call `connectedWithNewToken`, can't we implement it so it's automatically called? It seems like something that's easily forgettable. We can make `RetryRequester` a class (instead of a protocol) and it can accept `reconnect` closure (just like `tokenProvider` and `touchTokenProvider`)
* The extension function `connectedWithNewToken` having access to `Client.shared` directly is making it hard to test, can we start dependency injection?